numpy/_core/numerictypes.py

```diff
@@ -359,6 +360,90 @@ def issubsctype(arg1, arg2):
     return issubclass(obj2sctype(arg1), obj2sctype(arg2))
 
 
+@set_module('numpy')
+def isdtype(dtype, kind):
+    """
+    Returns a boolean indicating whether a provided dtype is of a specified
+    data type ``kind``. This function only supports Array API data types.
+
+    Parameters
+    ----------
+    dtype : dtype
+        The input dtype.
+    kind : dtype or str or tuple of dtypes/strs.
+        dtype or dtype kind. Allowed dtype kinds are:
+        * ``bool`` : boolean kind
+        * ``signed integer`` : signed integer data types
+        * ``unsigned integer`` : unsigned integer data types
+        * ``integral`` : integer data types
+        * ``real floating`` : real-valued floating-point data types
+        * ``complex floating`` : complex floating-point data types
+        * ``numeric`` : numeric data types
+
+    Returns
+    -------
+    out : bool
+
+    See Also
+    --------
+    issubdtype
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> np.isdtype(np.float32, np.float64)
+    False
+    >>> np.isdtype(np.int64, (np.uint64, "signed integer"))
+    True
+
+    """
+    # validate and preprocess arguments
```

Comment:
It would be useful to extend this comment to state what is actually happening here. The intended inputs are `np.float32` & co, which aren't instances of `multiarray.dtype`. So I'm not sure that this validation is right.

Also, rejection of non-compliant objects may be good to do more explicitly, now the errors may be a bit obscure:
```python
In [9]: np.isdtype(np.float32, np.ones(1))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[9], line 1
----> 1 np.isdtype(np.float32, np.ones(1))

File ~/code/numpy/build-install/usr/lib/python3.11/site-packages/numpy/_core/numerictypes.py:436, in isdtype(dtype, kind)
    434 if isinstance(kind, ma.dtype):
    435     kind = kind.type
--> 436 if kind not in allTypes.values():
    437     raise TypeError(
    438         "kind argument must be comprised of NumPy dtypes or "
    439         f"strings only, but it is a {kind}."
    440     )
    442 processed_kinds.add(kind)

TypeError: descriptor '__array_wrap__' for 'numpy.generic' objects doesn't apply to a 'numpy.ndarray' object
```